dbms/src/Storages/ColumnsDescription.cpp

```diff
@@ -421,4 +426,28 @@ ColumnsDescription ColumnsDescription::parse(const String & str)
     return result;
 }
 
+
+Block validateColumnsDefaultsAndGetSampleBlock(ASTPtr default_expr_list, const NamesAndTypesList & all_columns, const Context & context)
+{
+    for (const auto & child : default_expr_list->children)
```

Comment:
@4ertus2  I think this check is enough, but maybe I have missed something.

Reply:
Generally ASTSelectQuery appears inside of ASTSelectWithUnionQuery. So we'd better add both or none of them in check.